datastore/gcloud/aio/datastore/value.py

```diff
@@ -60,7 +64,7 @@ def from_repr(cls, data: Dict[str, Any]) -> 'Value':
 
     def to_repr(self) -> Dict[str, Any]:
         value_type = self._infer_type(self.value)
-        if value_type in {TypeName.GEOPOINT, TypeName.KEY}:
+        if value_type in {TypeName.ENTITY, TypeName.GEOPOINT, TypeName.KEY}:
```

Comment:
Could you explain this change? I think you _might_ have meant to add `TypeName.ARRAY` here rather than `ENTITY`?